Refit/RequestBuilderImplementation.cs

```diff
@@ -543,7 +543,21 @@ async Task AddMultipartItemAsync(MultipartFormDataContent multiPartContent, stri
                     // if header, add to request headers
                     if (restMethod.HeaderParameterMap.ContainsKey(i))
                     {
-                        headersToAdd[restMethod.HeaderParameterMap[i]] = paramList[i]?.ToString();
+                        var headerInfo = restMethod.HeaderParameterMap[i];
+                        if (headerInfo.IsHeaderCollection)
```

Comment:
Why is this check required? Why can't it just check for the `HeaderCollection` attribute on the parameter info?

Reply:
Please see the explain above:

1. Build a Position (int) -> Header Name (string) map from parameter list. All parameter with Header attribute will be picked up and added to the map.
1. Use this map to see if a parameter is in the map. If so, use the header name in the map, then use the parameter as the value.

Here in the step 2, the `HeaderParameterMap` is just a `IDictionary<string, string>` and it doesnot contains any attribute info - it's been built in step 1, if we get attribute info here then it's meaning less to build such a map in step 1.

Since I'm not familiar with this project, so I'll try to minimize the change. If you think remove the first step and get attribute info in step 2 in acceptable, I can do that as well.

Reply:
I'd prefer to do the "correct" change rather than a workaround. If the code that builds the map needs to change to do something different, then I think that's the better approach.

Reply:
Yes, that's what I actually did - I changed the map from `IDictionary<string, string>` to `IDictionary<string, HeaderInfo>` to handle more complex cases like `HeaderCollection`. In fact, I reviewed the code and found it's hard to get attribute info in step 2: this map is contained in a `RestMethodInfo` class which is built from C#'s native `MethodInfo`, and there's no attribute info in this `RestMethodInfo` class, so we cannot get attribute info in step 2 actually.

Reply:
Ping - @onovotny Any feedback? BTW, this two-steps implementation is not implemented by me, but exist here before I made the change - I'm just improving it, so it's not a **workaround**.